src/main/java/org/commcare/formplayer/services/MediaHandler.kt

```diff
@@ -19,16 +21,37 @@ class MediaHandler(val file: MultipartFile) {
     /**
      * Saves file in the given parent directory
      */
-    fun saveFile(parentDirPath: Path): String {
+    fun saveFile(parentDirPath: Path, session: SerializableFormSession, username: String, asUser: String,
+                 domain: String, appId: String): String {
         validateFile()
         val fileId = PropertyUtils.genUUID()
         val parent = parentDirPath.toFile()
         parent.mkdirs()
         var fileIdWithExt = fileId
-        FileUtils.getExtension(file.originalFilename)?.let { fileIdWithExt = "$fileId.$it" }
-        val desintationFile = getMediaFilePath(parentDirPath, fileIdWithExt).toFile()
+        val fileExtension = FileUtils.getExtension(file.originalFilename)
+        fileExtension?.let { fileIdWithExt = "$fileId.$it" }
+        val filePath = getMediaFilePath(parentDirPath, fileIdWithExt)
+        val destinationFile = filePath.toFile()
+
+        val metaDataRecordToDelete = MediaMetaDataService().findByFilePath(filePath.toString())
+        val mediaMetaDataService = MediaMetaDataService()
+        metaDataRecordToDelete?.let{ mediaMetaDataService.deleteMetaDataById(it.id) }
```

Comment:
since this `filePath` is a new path generated above with a new fileId - `PropertyUtils.genUUID()` , we should not have any records at the moment for this in the meta data table making it pointless to perform a deletion here. Instead we wanna do the deletion when we delete a particular file from filesystem as a result of user deleting/replacing any file from a question in web apps. This deletion happens in `cleanMedia` method of `MediaHandler` which is where we should add this code to delete the file metadata record as well.

Reply:
Got it, I'll move this logic to `cleanMedia`. What I was thinking here was that if there was already a metadata record with the file path that is being constructed here, we'd find that record with that file path and remove it.

Reply:
yeah that makes sense, though my sense is the chances of 2 files having the same `uuid` are practically close to near impossible i.e `PropertyUtils.genUUID()` should give us a near unique number.